shared-libs/search/src/search.js

```diff
@@ -74,15 +74,22 @@ module.exports = function(Promise, DB) {
     return queryView(request);
   };
 
-  var getRows = function(type, requests, options) {
+  var getRows = function(type, requests, options, returnExtendedResults) {
     if (requests.length === 1 && requests[0].ordered) {
       // 1 ordered view - let the db do the pagination for us
       return queryViewPaginated(requests[0], options);
     }
     // multiple requests - have to manually paginate
+    var extendedResults;
     return Promise.all(requests.map(queryView))
       .then(getIntersection)
-      .then(_.partial(getPageRows, type, _, options));
+      .then(function(results) {
+        extendedResults = results;
```

Comment:
Nit picky, but the name extendedResults suggests to me that it would be the result of an additional lookup. I'd almost call this intermediate results or something, since we are returning the result of something we get for free but typically throw away.

Similarly, nit picky -- maybe something like "extendedResultsCallback" instead of "returnExtendedResults" for the parameter name.

Reply:
Or `queryResultsCache` or something?

Reply:
`queryResultsCache` it is!